wordpress-mobile/WordPress-Android
# Issue/10632 material site picker

Fixes #10632 

This PR moves Site Picker screen to Material Theme with Dark Theme support. I also had to do a change of SearchView logic that should not affect functionality.

This PR affects other screens too and they might look broken until we completely move to Material Theme.

PS: I borrowed some hints on the styling of Action Mode from Simplenote :)

ping @mattmiklic 
![comparison](https://user-images.githubusercontent.com/728822/68539239-1ba26900-0335-11ea-8728-a402943a0d60.png)

To test:
Apart from visual changes, there were some minor changes around Search and site selection functionality.
- Try Site picker with Light and Dark themes. The theme can be changed from App Settings.
- Make sure the Search is working as expected.
- Make sure long-pressing on a site transitions to selection mode, where sites can be selected.


PR submission checklist:
